src/smt/smt_engine.cpp

```diff
@@ -4862,33 +4873,39 @@ void SmtEngine::checkSynthSolution()
     Node conj = Node::fromExpr(*i);
     // Apply any define-funs from the problem.
     {
-      unordered_map<Node, Node, NodeHashFunction> cache;
+      std::unordered_map<Node, Node, NodeHashFunction> cache;
       conj = d_private->expandDefinitions(conj, cache);
```

Comment:
We don't expand definitions when doing SyGuS, is that this is necessary? For all assertions?

Reply:
Expand definitions is always called on all assertions (https://github.com/CVC4/CVC4/blob/master/src/smt/smt_engine.cpp#L3241) regardless of what kind of check is run.

Reply:
So doing it again here is redundant?

Reply:
No, it is run on assertions prior to giving them to the solver. However, we maintain the assertions in their original form because they are the ground truth of what the input was.  Thus, we redo it here again.  This is analogous to the code for `check-model`.

Reply:
I see. Didn't remember that. :)